src/renderer/utils/TestSequenceProjectHandler.ts

```diff
@@ -0,0 +1,335 @@
+  }
+}
+
+function validatePath(project: TestSequencerProject): TestSequencerProject {
+  let path = project.projectPath;
+  if (!path.endsWith("/")) {
+    path = path + "/";
+  }
+  return updatePath(project, path);
+}
+
+function updatePath(
+  project: TestSequencerProject,
+  newPath: string,
+): TestSequencerProject {
+  return {
+    ...project,
+    projectPath: newPath,
+  };
+}
+
+async function saveToDisk(project: TestSequencerProject): Promise<void> {
+  if ("api" in window) {
+    // Deps
+    if (project.interpreter.requirementsPath) {
+      const deps = await window.api.poetryShowUserGroup();
+      const content = deps
+        .map((dep) => dep.name + "==" + dep.version)
+        .join("\n");
+      await window.api.saveToFile(
+        project.projectPath + project.interpreter.requirementsPath,
+        content,
+      );
+    }
+    // Project
+    await window.api.saveToFile(
+      project.projectPath + project.name + ".tjoy",
+      stringifyProject(project),
+    );
+  } else {
+    throw new Error("Not able to save to disk");
+  }
+}
+
+async function installDeps(project: TestSequencerProject): Promise<boolean> {
+  if ("api" in window) {
+    const succes = await window.api.poetryInstallRequirementsUserGroup(
+      project.projectPath + project.interpreter.requirementsPath,
+    );
+    return succes;
+  } else {
+    throw new Error("Not able to install requirements."); // TODO: Better error
+  }
+}
+
+async function syncProject(
+  project: TestSequencerProject,
+  stateManager: StateManager,
+): Promise<void> {
+  const elements = await createTestSequencerElementsFromProjectElements(
+    project,
+    project.projectPath,
+    false,
+  );
+  stateManager.setElems(elements);
+  stateManager.setProject(project);
+  stateManager.setUnsaved(false);
+}
+
+async function createProjectElementsFromTestSequencerElements(
+  elems: TestSequenceElement[],
+  baseFolder: string,
+  verifStateOrThrow: boolean,
+): Promise<TestSequenceElement[]> {
+  baseFolder = baseFolder;
+  if (verifStateOrThrow) {
+    await throwIfNotInAllBaseFolder(elems, baseFolder);
+  }
+  const elements = [...elems].map((elem) => {
+    return elem.type === "test"
+      ? {
+          ...elem,
+          status: "pending",
+          completionTime: undefined,
+          error: null,
+          isSavedToCloud: false,
+          testName: elem.path.replaceAll(baseFolder, ""),
+          path: elem.path.replaceAll(baseFolder, ""),
+        }
+      : {
+          ...elem,
+          condition: elem.condition.replaceAll(baseFolder, ""),
+        };
+  });
+  // @ts-ignore because LSP can't understand the element.type
+  return elements;
+}
+
+async function createTestSequencerElementsFromProjectElements(
+  project: TestSequencerProject,
+  baseFolder: string,
+  verifStateOrThrow: boolean,
+): Promise<TestSequenceElement[]> {
+  const elements: TestSequenceElement[] = [...project.elems].map((elem) => {
+    return elem.type === "test"
+      ? {
```

Comment:
if possible, use the util function I mentioned to do above createNewTest() (pretty easy with zustand too)